--- origsrc/libcdio-0.90/configure.ac	2012-10-27 10:07:45.000000000 -0500
+++ src/libcdio-0.90/configure.ac	2013-11-20 15:13:21.580067500 -0600
@@ -443,8 +443,8 @@ int has_timeout=sizeof(test.timeout);],
 	## Cross-platform LFS support in MinGW requires an override of off_t
 	if test "x$enable_largefile" != "xno"; then
 	  AC_MSG_RESULT([enabling support for large files (_FILE_OFFSET_BITS=64)])
-	  LIBCDIO_CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_OFF_T_DEFINED -D_OFF_T_ -D_off_t=off64_t -Doff_t=off64_t'
-	  LIBISO9660_CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_OFF_T_DEFINED -D_OFF_T_ -D_off_t=off64_t -Doff_t=off64_t'
+	  LIBCDIO_CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_FILE_OFFSET_BITS_SET_LSEEK'
+	  LIBISO9660_CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_FILE_OFFSET_BITS_SET_LSEEK'
 	fi
 	joliet_supported="yes"
 	if test "x$enable_cdda_player" = "xyes"; then
--- origsrc/libcdio-0.90/include/cdio/udf_time.h	2012-03-03 10:47:36.000000000 -0600
+++ src/libcdio-0.90/include/cdio/udf_time.h	2013-11-20 15:15:47.531415400 -0600
@@ -27,7 +27,7 @@
 
 #include <time.h>
 
-#if defined(__MINGW32__) && !defined(__MINGW64__)
+#if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR)
 struct timespec {
   time_t  tv_sec;   /* Seconds */
   long    tv_nsec;  /* Nanoseconds */
--- origsrc/libcdio-0.90/lib/driver/MSWindows/win32_ioctl.c	2012-10-27 10:07:45.000000000 -0500
+++ src/libcdio-0.90/lib/driver/MSWindows/win32_ioctl.c	2013-11-20 15:17:14.639397700 -0600
@@ -31,7 +31,7 @@
 # include "NtScsi.h"
 # include "undocumented.h"
 #else
-# if defined (__MINGW64__)
+# if defined (__MINGW64_VERSION_MAJOR)
 #  define _NTSRB_ /* Bad things happen if srb.h gets included */
 # endif
 # include <windows.h>
--- origsrc/libcdio-0.90/lib/driver/_cdio_generic.c	2012-10-24 20:33:54.000000000 -0500
+++ src/libcdio-0.90/lib/driver/_cdio_generic.c	2013-11-20 15:09:12.356812700 -0600
@@ -57,9 +57,7 @@
 
 /* If available and LFS is enabled, try to use lseek64 */
 #if defined(HAVE_LSEEK64) && defined(_FILE_OFFSET_BITS) && (_FILE_OFFSET_BITS == 64)
-#if defined(_MSC_VER)
 #include <io.h>
-#endif
 #define CDIO_LSEEK lseek64
 #else
 #define CDIO_LSEEK lseek
